pkg/resourcemanager/controller/managedresource/sort.go

```diff
@@ -57,3 +58,60 @@ func (r referenceSorter) Swap(i, j int) {
 	r.keys[i], r.keys[j] = r.keys[j], r.keys[i]
 	r.refs[i], r.refs[j] = r.refs[j], r.refs[i]
 }
+
+type kindSorter struct {
+	ordering map[string]int
+	objects  []object
+}
+
+func newKindSorter(obj []object, s SortOrder) *kindSorter {
+	o := make(map[string]int, len(s))
+	for v, k := range s {
+		o[k] = v
+	}
+
+	return &kindSorter{
+		objects:  obj,
+		ordering: o,
+	}
+}
+
+func (k *kindSorter) Len() int { return len(k.objects) }
+
+func (k *kindSorter) Swap(i, j int) { k.objects[i], k.objects[j] = k.objects[j], k.objects[i] }
+
+func (k *kindSorter) Less(i, j int) bool {
+	a := k.objects[i]
+	b := k.objects[j]
+	first, aok := k.ordering[a.oldInformation.Kind]
+	second, bok := k.ordering[b.oldInformation.Kind]
+
+	if !aok && !bok {
+		// if both are unknown then sort alphabetically by kind and name
+		if a.oldInformation.Kind != b.oldInformation.Kind {
+			return a.oldInformation.Kind < b.oldInformation.Kind
+		}
+		return a.oldInformation.Name < b.oldInformation.Name
+	}
+
+	// unknown kind is last
+	if !aok {
+		return false
+	}
+	if !bok {
+		return true
+	}
+
+	// if same kind sub sort alphanumeric
+	if first == second {
+		return a.oldInformation.Name < b.oldInformation.Name
+	}
+	// sort different kinds
+	return first < second
+}
+func sortByKind(resourceObject []object) []object {
+	ordering := InstallOrder
```

Comment:
can we not copy the InstallOrder to this package also?

Reply:
I am using that only

Reply:
I was suggesting to copy it instead of importing.